### PR TITLE
3.5 into 3.6

### PR DESCRIPTION
**Pull request by jack-w-shaw**
Forward merge:
- https://github.com/juju/juju/pull/17348
- https://github.com/juju/juju/pull/17413
- https://github.com/juju/juju/pull/17423
- https://github.com/juju/juju/pull/17436
- https://github.com/juju/juju/pull/17422
- https://github.com/juju/juju/pull/17446
- https://github.com/juju/juju/pull/17448
- https://github.com/juju/juju/pull/17444
- https://github.com/juju/juju/pull/17453
- https://github.com/juju/juju/pull/17440
- https://github.com/juju/juju/pull/17458

Conflicts:
- acceptancetests/jujupy/client.py
- acceptancetests/jujupy/tests/test_status.py
- acceptancetests/repository/trusty/haproxy/cm.py
- acceptancetests/tests/test_jujucharm.py
- cmd/juju/cloud/addcredential.go
- core/base/base.go
- environs/bootstrap/bootstrap.go
- provider/azure/environ.go
- provider/azure/internal/imageutils/images.go
- scripts/win-installer/setup.iss
- snap/snapcraft.yaml
- version/version.go

Conflicts mostly trivial

`provider/azure/internal/imageutils/images.go` took a bit of thought. This conflict resulted from my PR https://github.com/juju/juju/pull/17440

### CODE REVIEW

File: cmd/juju/application/bundle/bundle.go [hunk 1]
```diff
@@ -355,6 +358,29 @@ func verifyBundle(data *charm.BundleData, bundleDir string, verifyConstraints fu
...
+	}
+	for _, app := range data.Applications {
+		if app != nil && app.Series != "" {
+			includeSeries = true
+			break
+		}
+	}
+
+	if includeSeries {
+		ctx.Warningf("series in being deprecated in favour of bases. For more information about the transition to bases see https://discourse.charmhub.io/t/transition-from-series-to-base-in-juju-4-0/14127")
```
Comment (manadart) [inline cmd/juju/application/bundle/bundle.go:380]: Is this a typo @jack-w-shaw ?

**Review by manadart — APPROVED** [review]
Might have to go back and fix a string.

File: cmd/juju/application/bundle/bundle.go (hunk 1 above)
Comment (jack-w-shaw) [reply cmd/juju/application/bundle/bundle.go:380]: Damn it is

**Pull request merged**